app/src/main/java/org/mifos/mobile/viewModels/RecentTransactionViewModel.kt

```diff
@@ -28,24 +28,34 @@ class RecentTransactionViewModel @Inject constructor(private val recentTransacti
         loadRecentTransactions(offset, limit)
     }
 
-    private fun loadRecentTransactions(offset: Int, limit: Int) {
+    private fun loadRecentTransactions(offset: Int, limit: Int, refresh: Boolean = false) {
         viewModelScope.launch {
-            _recentTransactionUiState.value = RecentTransactionUiState.Loading
-            recentTransactionRepositoryImp.recentTransactions(offset, limit).catch {
-                _recentTransactionUiState.value =
-                    RecentTransactionUiState.Error(R.string.recent_transactions)
-            }.collect {
-                if (it.totalFilteredRecords == 0) {
-                    _recentTransactionUiState.value = RecentTransactionUiState.EmptyTransaction
-                } else if (loadmore && it.pageItems.isNotEmpty()) {
-                    _recentTransactionUiState.value =
-                        RecentTransactionUiState.LoadMoreRecentTransactions(it.pageItems)
-                } else if (it.pageItems.isNotEmpty()) {
+            try {
```

Comment:
why are you using try block ? catch api doesn't need to work with try

Reply:
there has to be a better alternative

Reply:
Or else i will just remove Try and finally set the state at the end to initial (thats works same as finally) what's Your view...? 
 @PratyushSingh07